### ISSUE TITLE
[composer] Error dialog/reset

### ISSUE THREAD

**Issue by richburdon** [dxos/dxos#5726]
- Current RTE Error dialog gives option to reset all data (different less intrusive UX than red danger message in HALO).
- Reset doesn't work.
- Reset from HALO throws error below (then succeeds with additional worker error on restart)

```
react-DYBMIjzU.js:30 Uncaught Error: invariant violation: HaloProxy not opened [this._invitationProxy] at packages/sdk/client/src/halo/halo-proxy.ts:89
    at invariant (dxos-qbtehQOE.js:10:45562)
    at get invitations (halo-qzZXyFM8.js:1:1893)
    at index-jGNUWLw9.js:1:838
    at Object.ji [as useSyncExternalStore] (react-DYBMIjzU.js:30:21576)
    at P.useSyncExternalStore (react-DYBMIjzU.js:9:6419)
    at o (index-jGNUWLw9.js:1:729)
    at Ea (react-iN7XjfoY.js:3:46001)
    at ql (react-DYBMIjzU.js:30:19478)
    at no (react-DYBMIjzU.js:32:3134)
    at Qs (react-DYBMIjzU.js:32:44467)
```

**Comment by wittjosiah** [comment]
@richburdon it's unclear to me what the action of this issue is intended to be. What is the desired outcome? Are you suggesting the reset option be removed? UI changed? Simply fix the functionality? Also unsure what the code blocks are, how to reproduce them, or what "Reset from HALO" means.

**Comment by zhenyasav** [comment]
by HALO he actually means "shell"

**Comment by wittjosiah** [comment]
potential fix for reset here https://github.com/dxos/dxos/pull/5792 haven't verified yet

**Comment by zhenyasav** [comment]
There was some discussion about whether the reset experience needs to be one and the same between what's in the ErrorBoundary (RTE dialog) and what's in the shell panels. We landed on the idea that those two are indeed different and should not be the same - one is a fallback and tries to minimize the chance of error inside the fallback, and the other is a more complex (and error prone) piece of code for all 1st and 3rd party apps, and should not figure in the critical error handling fallback due to it's complexity. In other words, the button in the ErrorBoundary should just call the sdk reset api and should itself be minimal and self-contained, without shell.

IMO what to do here:
- [ ] ensure a test that reset works from the error-boundary 
- [ ] ensure a test that reset works from a shell panel

**Comment by wittjosiah** [comment]
We do have a test for reset from shell already so we've got that covered. Agreed on adding one for this path, I realized that gap when looking into this the first time. If the above PR did fix it I will add a test to cover it, if not I'll add one once it's fixed.

In terms of what is happening in the error boundary, you're right that it's different. In the error boundary we aren't guaranteed to have an initialized client, so it attempts a minimal startup of some backend services in order to simple clear storage and reload the app.
